src/bp/ui-studio/src/web/components/Shared/Interface/index.tsx

```diff
@@ -0,0 +1,185 @@
+import {
+  Button,
+  ButtonGroup,
+  Classes,
+  Collapse,
+  ControlGroup,
+  Icon,
+  InputGroup,
+  Popover,
+  Position,
+  Tooltip
+} from '@blueprintjs/core'
+import classnames from 'classnames'
+import React, { useState } from 'react'
+import { HotKeys } from 'react-hotkeys'
+import { MdHelpOutline, MdInfoOutline } from 'react-icons/md'
+import SplitPane from 'react-split-pane'
+
+import style from './style.scss'
+import {
+  ContainerProps,
+  InfoTooltipProps,
+  ItemListProps,
+  KeyboardShortcutsProps,
+  SearchBarProps,
+  SectionAction,
+  SectionProps,
+  SplashScreenProps
+} from './typings'
+
+import { buildMenu } from './utils'
+
+export const Container = (props: ContainerProps) => {
+  const [sidebarVisible, setSidebarVisible] = useState(!props.sidebarHidden)
+  const width = props.sidebarWidth ? props.sidebarWidth : 300
+
+  const toggleSidebar = () => setSidebarVisible(!sidebarVisible)
+  window.toggleSidebar = toggleSidebar
+
+  const keyHandlers = {
+    'toggle-sidebar': toggleSidebar,
+    ...(props.keyHandlers || {})
+  }
+
+  return (
+    <HotKeys handlers={keyHandlers} focused style={{ width: '100%', height: '100%' }}>
+      <div className={classnames(style.container, { [style.sidebar_hidden]: !sidebarVisible })}>
+        <SplitPane split="vertical" defaultSize={width} size={sidebarVisible ? width : 0}>
+          {props.children}
+        </SplitPane>
+      </div>
+    </HotKeys>
+  )
+}
+
+export const Section = (props: SectionProps) => {
```

Comment:
Does this notation gives the typings intellisens when we use the component ?

If not you might have to type your components as 
```tsx

export const Section: React.FC<SectionProps> = props => {
 //...
}
```

Reply:
I might like if you called this SideBarSection just because it is more explicit and that we might end up with Section component for the content or the eventual bottom panel.

Reply:
Good for the naming. And yes you get intellisense when you are in the studio. Don't know how to enable it for webpack externals tho. I think we would need to import the typings into separate modules